runbotics/runbotics-orchestrator-ui/src/src-app/views/process/EditProcessDialog.tsx

```diff
@@ -29,71 +33,147 @@ type EditProcessDialogProps = {
     onAdd: (process: IProcess) => void;
 };
 
+export interface FormValidationState {
+    name: boolean;
+}
+
+const initialFormValidationState: FormValidationState = {
+    name: true
+};
+
+const getProcessDataWithoutNulls = (process: IProcess): IProcess => ({
+    ...process,
+    description: process.description ?? ''
+});
+
+const getProcessDataWithoutEmptyStrings = (process: IProcess): IProcess => ({
+    ...process,
+    description: (process.description === '' ? null : process.description)
+});
+
 const EditProcessDialog: FC<EditProcessDialogProps> = ({
     process, onAdd, onClose, open,
 }) => {
-    const [formState, setFormState] = useState<IProcess>({ ...process });
     const { translate } = useTranslations();
+    const dispatch = useDispatch();
 
-    const handleSubmit = () => {
-        onAdd(formState);
+    const [formValidationState, setFormValidationState] = useState<FormValidationState>(initialFormValidationState);
+    const [processFormState, setProcessFormState] = useState<IProcess>();
+    const [searchedDatabaseTags, setSearchedDatabaseTags] = useState<Tag[]>([]);
+    const [selectedTagsNames, setSelectedTagsNames] = useState<string[]>([]);
+    const [search, setSearch] = useState<string>('');
+    const debouncedSearch = useDebounce<string>(search.trim(), DEBOUNCE_TIME);
+
+    const handleTagChange = (updatedTags: string[]) => {
+        setSelectedTagsNames(updatedTags);
+
+        const hasTagBeenAdded = updatedTags.length > selectedTagsNames.length;
+        if (hasTagBeenAdded) {
+            const tagToAdd = (searchedDatabaseTags.find(tag => tag.name === updatedTags.at(-1))) ?? { name: updatedTags.at(-1) };
+            setProcessFormState((prevState) => ({
+                ...prevState,
+                tags: [...processFormState.tags, tagToAdd]
+            }));
+        } else {
+            const filteredTags = [...processFormState.tags].filter((tag) => updatedTags.includes(tag.name));
+            setProcessFormState((prevState) => ({
+                ...prevState,
+                tags: filteredTags
+            }));
+        }
+
+        setSearch('');
+    };
+
+    const handleTagSearch = (e: ChangeEvent<HTMLInputElement>) => {
+        const filteredSearch = e.target.value.replaceAll(/[^A-Za-z0-9]/g, '');
+        if (filteredSearch.length <= MAX_TAG_LENGTH) setSearch(filteredSearch);
```

Comment:
what if filteredSearch.length exceeds MAX_TAG_LENGTH? nothing happens?

Reply:
You can still write, but input saves your last 20 characters, and you can still add that string, it doesn't include the rest while adding that tag.
![image](https://github.com/runbotics/runbotics/assets/80831299/a791d371-41d5-472b-8ec9-89f049e74674)